scripts/zones/Windurst_Waters_[S]/npcs/Hampu-Kampu.lua

```diff
@@ -0,0 +1,54 @@
+-----------------------------------
+-- Area: Windurst Waters (S)
+--  NPC: Hampu-Kampu
+-- Type: Quest NPC
+-----------------------------------
+local ID = require("scripts/zones/Windurst_Waters_[S]/IDs")
+require("scripts/globals/quests");
+require("scripts/globals/keyitems");
+
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local sayItWithAHandbag = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if (sayItWithAHandbag == QUEST_COMPLETED) then
+        player:startEvent(175)
+    elseif (player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG) and sayItWithAHandbagCS == 4) then
+        player:startEvent(174)
+        print("repaired bag and 4")
+    elseif (player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) or sayItWithAHandbagCS == 3) then
+        player:startEvent(173)
+        print(player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG))
+    elseif (sayItWithAHandbag == QUEST_ACCEPTED and sayItWithAHandbagCS == 1) then
+        player:startEvent(172)
+    elseif (sayItWithAHandbag == QUEST_ACCEPTED) then
+        player:startEvent(170)
+    elseif (sayItWithAHandbag == QUEST_AVAILABLE) then
+        player:startEvent(169)
+        player:setVar("sayItWithAHandbagCS",0)
```

Comment:
You don't need to _initialize_ player vars at zero. Only set them to zero to delete them when you're done.

(And make sure not to do any setting in onTriggers, only eventFinishes!)

Reply:
thank you. 🙏

 OMG, can't believe I put that inside the onTrigger. Will make sure to spot check better 🙀